packages/authenticator/amplify_authenticator/test/authenticated_view_test.dart

```diff
@@ -29,35 +38,62 @@ void main() {
       await tester.sendKeyEvent(LogicalKeyboardKey.tab);
 
       // Enter text to currently focused widget (Username TextField).
-      tester.testTextInput.enterText('user@example.com');
+      tester.testTextInput.enterText(username);
 
       // Move focus to next widget via keyboard (Password TextField).
       await tester.sendKeyEvent(LogicalKeyboardKey.tab);
 
       // Enter text to currently focused widget (Password TextField).
-      tester.testTextInput.enterText('Password123');
+      tester.testTextInput.enterText(password);
 
-      expect(mockState.username, 'user@example.com');
-      expect(mockState.password, 'Password123');
+      expect(mockState.username, username);
+      expect(mockState.password, password);
 
-      // Move focus to first next widget via keyboard (Password Hide/Show toggle).
-      await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+      if (submitViaTextField) {
+        // Submit form via Enter from password field.
+        await tester.testTextInput.receiveAction(TextInputAction.done);
+      } else {
+        // Move focus to first next widget via keyboard (Password Hide/Show toggle).
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
 
-      // Move focus to first next widget via keyboard (Sign In Button).
-      await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        // Move focus to first next widget via keyboard (Sign In Button).
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
 
-      verifyNever(mockState.signIn);
+        // Submit form with Enter key.
+        await tester.sendKeyEvent(LogicalKeyboardKey.enter);
```

Comment:
Would tapping the button have the same effect?

Reply:
Okay, sounds good. I wouldn't mind it as a base case in these tests, but every other test in the package pretty much uses the button so I think it's okay.